lib/bolt/task.rb

```diff
@@ -54,9 +54,12 @@ def select_implementation(target, additional_features = [])
                files.first.dup
              end
 
-      unless (inmethod = metadata['input_method']).nil?
+      unless (inmethod = impl['input_method'] || metadata['input_method']).nil?
```

Comment:
```inmethod = impl['input_method'] || metadata['input_method']
impl['input_method'] = inmethod unless inmethod.nil?
```
is easier to read IMO
but what happens if the key is present but nil? ie
`impl['input_method'] ||= metadata['input_method']`

Reply:
I didn't want to add a nil entry if neither had it. But this is probably never printede...